packages/contracts/controller/src/account.cairo

```diff
@@ -425,8 +449,14 @@ mod CartridgeAccount {
         fn is_valid_span_signature(
             self: @ContractState, hash: felt252, signer_signatures: Span<SignerSignature>
         ) -> bool {
-            assert(signer_signatures.len() <= 2, 'invalid-signature-length');
-            self.is_valid_owner_signature(hash, *signer_signatures.at(0))
+            if signer_signatures.is_empty() {
+                return false;
+            }
+            if signer_signatures.len() > 2 {
+                return false;
+            }
+            self.is_valid_owner_signature(hash, *signer_signatures.at(0)) &&
```

Comment:
can we do `assert(signer_signatures.len() == 2, ...)` and have the dummy guardian sig that isn't validated if necessary?

Reply:
why would we require a guardian signature when there is no guardian on the account? It think this would actually hit performance more than the optionable storage, since we are deserializing the signatures.

Reply:
fair enough